File: downloads/code/clj/fibo.clj
(def fibo-seq
  (map first 
    (iterate (fn [[a b]] [b (+ a b)]) [0N 1N])))

(defn fibo-list [n]
  (take n fibo-seq))

(defn fibo-last [n]
  (last (fibo-list n)))

File: downloads/code/scala/Fibo.scala
object Fibo {
  lazy val fibo: Stream[BigInt] =
    BigInt(0) #::
    BigInt(1) #::
    fibo.zip(fibo.tail).map { n => n._1 + n._2 }

  def fiboList = fibo.take(_: Int).toList

  def fiboLast = fiboList(_: Int).last
}
